src/bindfs.c

```diff
@@ -1041,47 +1038,52 @@ static int bindfs_access(const char *path, int wants)
  *
  * @param path The path to iterate over.
  *
- * @return 0 if all path components have the search permission, -EACCES
+ * @return true if all path components have the search permission, false
  *     otherwise.
  */
-static int check_path_search_perms(const char *path) {
-    size_t path_length = strlen(path);
-    char *part = malloc(path_length + 1);
-    memcpy(part, path, path_length + 1);
+static bool path_has_search_perms(const char *path) {
+    char *part = strdup(path);
 
-    int access_check_result;
+    bool parent_dir = true;
     do {
         part = dirname(part);
+
+        if (parent_dir && access_check(part, W_OK) != 0) {
```

Comment:
Write permission? Doesn't look right.

Reply:
Thanks, well spotted. I merged the `dirname_access_check` into this function and then (apparently only partially) undid the changes.